app/src/userpages/components/StreamPage/Show/KeyView/index.jsx

```diff
@@ -47,9 +49,19 @@ export class KeyView extends Component<Props> {
         }
     }
 
-    getKeys = () => {
-        if (this.props.disabled || this.props.streamId == null) { return }
-        this.props.getKeys(this.props.streamId)
+    getKeys = async () => {
+        const { disabled, streamId, getKeys } = this.props
+
+        if (disabled || streamId == null) { return }
+        try {
+            await getKeys(streamId)
```

Comment:
Is it possible that `this.props.streamId` (or even `disabled`) is different by the time this resolves? i.e. might it show a failure notification for the wrong `streamId`?

Reply:
Fixed in cd0780dfbce808889fd5565320179db14f2a32c2.

Reply:
The iead here is that each time we call `getKeys` we ignore failures from the previous calls. FYI.